### PR TITLE
Update storybook documentation and add introductory page

### PR DESCRIPTION
**Pull request by louwie17**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR makes a couple changes:
- Adds an `introduction` page to the storybook, similar to that of Gutenberg ( see [example](https://wordpress.github.io/gutenberg/?path=/docs/docs-introduction--page) )
- Adds a small paragraph about storybook to the ux guidelines README
- Update the prefixes of the storybook names, as these were not overly consistent. Now they are aligned with the package names.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Read through the README changes and make sure it makes sense.
2. Run `pnpm --filter=@woocommerce/storybook watch:build` and go to `http://localhost:6007/`
3. Make sure it loads the introduction page first
4. Check if the left side bar corresponds well with the package names.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: tools/storybook/stories/docs/introduction.mdx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{/* introduction.mdx */}
```
Comment (gigitux) [inline tools/storybook/stories/docs/introduction.mdx]: Should we keep this comment?

**Review by gigitux — APPROVED** [review]
Thanks for working on this! I left a minor comment, but LGTM!

Is it expected that “WooCommerce Blocks” be in another section?

image

I would move them to the same level of others, and renaming “WooCommerce Blocks” with “Blocks”

File: tools/storybook/stories/docs/introduction.mdx (hunk 1 above)
Comment (louwie17) [reply tools/storybook/stories/docs/introduction.mdx]: Yeah I can probably remove that, it was copy/paste from what `core` did

**Comment by louwie17** [comment]
> Is it expected that “WooCommerce Blocks” be in another section?

Yeah, currently WooCommerce Blocks is still its own storybook, and I make use of `refs` to reference it: https://github.com/woocommerce/woocommerce/blob/63fc693003e3d6812ee9d2858e4723e940929013/tools/storybook/.storybook/main.js#L29
I am not sure if I can display it any differently, see the Storybook Composition docs: https://storybook.js.org/docs/sharing/storybook-composition

> I would move them to the same level of others, and renaming “WooCommerce Blocks” with “Blocks”

This is certainly possible, let me do that.

File: tools/storybook/stories/docs/introduction.mdx (hunk 1 above)
Comment (louwie17) [reply tools/storybook/stories/docs/introduction.mdx]: Removed: https://github.com/woocommerce/woocommerce/pull/52826/commits/00a862802a2eb152b8a98b7d3df9e3cdeda83e5d

**Comment by gigitux** [comment]
> Yeah, currently WooCommerce Blocks is still its own storybook, and I make use of `refs` to reference it:

Gotcha!

Thanks for addressing the feedback! :shipit:

**Pull request merged**